From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ricardo Pardini <ricardo@pardini.net>
Date: Sun, 4 Jan 2026 15:58:51 +0100
Subject: GENIO: radxa-nio-12l: Enable BTRFS and BZIP2 support

After cherry-picking from upstream, BTRFS should work, so enable it
for the NIO-12L. Also add support for general BZIP2 compression.

Signed-off-by: Ricardo Pardini <ricardo@pardini.net>
---
 configs/genio_1200_radxa_nio_12l_d16_defconfig | 2 ++
 configs/genio_1200_radxa_nio_12l_d4_defconfig  | 2 ++
 configs/genio_1200_radxa_nio_12l_d8_defconfig  | 2 ++
 3 files changed, 6 insertions(+)

diff --git a/configs/genio_1200_radxa_nio_12l_d16_defconfig b/configs/genio_1200_radxa_nio_12l_d16_defconfig
index 111111111111..222222222222 100644
--- a/configs/genio_1200_radxa_nio_12l_d16_defconfig
+++ b/configs/genio_1200_radxa_nio_12l_d16_defconfig
@@ -53,6 +53,7 @@ CONFIG_CMD_USB_MASS_STORAGE=y
 # CONFIG_CMD_ITEST is not set
 # CONFIG_CMD_BLOCK_CACHE is not set
 CONFIG_CMD_EFIDEBUG=y
+CONFIG_CMD_BTRFS=y
 CONFIG_PARTITION_TYPE_GUID=y
 CONFIG_ENV_IS_IN_SCSI=y
 CONFIG_SYS_SCSI_ENV_DEV=1
@@ -122,6 +123,7 @@ CONFIG_USB_ETHER=y
 CONFIG_UFS=y
 CONFIG_UFS_MEDIATEK=y
 CONFIG_WDT=y
+CONFIG_BZIP2=y
 CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_EFI_SET_TIME=y
 CONFIG_EFI_RUNTIME_UPDATE_CAPSULE=y
diff --git a/configs/genio_1200_radxa_nio_12l_d4_defconfig b/configs/genio_1200_radxa_nio_12l_d4_defconfig
index 111111111111..222222222222 100644
--- a/configs/genio_1200_radxa_nio_12l_d4_defconfig
+++ b/configs/genio_1200_radxa_nio_12l_d4_defconfig
@@ -53,6 +53,7 @@ CONFIG_CMD_USB_MASS_STORAGE=y
 # CONFIG_CMD_ITEST is not set
 # CONFIG_CMD_BLOCK_CACHE is not set
 CONFIG_CMD_EFIDEBUG=y
+CONFIG_CMD_BTRFS=y
 CONFIG_PARTITION_TYPE_GUID=y
 CONFIG_ENV_IS_IN_SCSI=y
 CONFIG_SYS_SCSI_ENV_DEV=1
@@ -122,6 +123,7 @@ CONFIG_USB_ETHER=y
 CONFIG_UFS=y
 CONFIG_UFS_MEDIATEK=y
 CONFIG_WDT=y
+CONFIG_BZIP2=y
 CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_EFI_SET_TIME=y
 CONFIG_EFI_RUNTIME_UPDATE_CAPSULE=y
diff --git a/configs/genio_1200_radxa_nio_12l_d8_defconfig b/configs/genio_1200_radxa_nio_12l_d8_defconfig
index 111111111111..222222222222 100644
--- a/configs/genio_1200_radxa_nio_12l_d8_defconfig
+++ b/configs/genio_1200_radxa_nio_12l_d8_defconfig
@@ -53,6 +53,7 @@ CONFIG_CMD_USB_MASS_STORAGE=y
 # CONFIG_CMD_ITEST is not set
 # CONFIG_CMD_BLOCK_CACHE is not set
 CONFIG_CMD_EFIDEBUG=y
+CONFIG_CMD_BTRFS=y
 CONFIG_PARTITION_TYPE_GUID=y
 CONFIG_ENV_IS_IN_SCSI=y
 CONFIG_SYS_SCSI_ENV_DEV=1
@@ -122,6 +123,7 @@ CONFIG_USB_ETHER=y
 CONFIG_UFS=y
 CONFIG_UFS_MEDIATEK=y
 CONFIG_WDT=y
+CONFIG_BZIP2=y
 CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_EFI_SET_TIME=y
 CONFIG_EFI_RUNTIME_UPDATE_CAPSULE=y
-- 
Armbian

